### ISSUE TITLE
`bun test` doesn't auto set `NODE_ENV=test`

### ISSUE THREAD

**Issue by hyuzea** [oven-sh/bun#4118]
### What version of Bun is running?

0.7.4

### What platform is your computer?

ubuntu 22.04

### What steps can reproduce the bug?

bun test

### What do you see instead?

this error is meant not to be printed in `NODE_ENV === test`, but see the current env in the bottom

![image](https://github.com/oven-sh/bun/assets/3823496/acd9833d-6ffd-4583-a1ab-fd3362589792)

**Comment by Jarred-Sumner** [comment]
This definitely used to work

**Comment by Galzk** [comment]
I think I've got a reliable repro:
```javascript
    // @@ Scenario 1: test true
    console.log("@@ Scenario 1:", process.env.NODE_ENV, process.env.NODE_ENV === "test");
    // @@ Scenario 2: undefined false
    console.log("@@ Scenario 2:", process?.env?.NODE_ENV, process?.env?.NODE_ENV === "test");
    // @@ Scenario 3: test true
    console.log("@@ Scenario 3:", process && process.env && process.env.NODE_ENV, process && process.env && process.env.NODE_ENV === "test");
```
Optional chaining is quite important as I check the var in another runtime

**Comment by wiredmatt** [comment]
What's the workaround? This ruins CI and even local development
```
❯ bun --version
1.0.15
```

`bun --env-file=.env.test src/index.ts` seems to get the job done.

**Comment by paperclover** [comment]
i dont really consider this to be a bug, but rather just using the existing environment variables. if NODE_ENV does not exist already, it is set to `test`

<img width="961" alt="image" src="https://github.com/oven-sh/bun/assets/24465214/42ea469a-7bf7-49a4-bda2-fe506af8bb08">

maybe this should force `test`, but to be noted that jest and vitest also observe this same behavior.

vitest

<img width="533" alt="image" src="https://github.com/oven-sh/bun/assets/24465214/c65abe58-1768-4de6-b2fb-c3a5d247b4b5">

jest

<img width="440" alt="image" src="https://github.com/oven-sh/bun/assets/24465214/5ebfae04-5252-4d42-b3f3-b924c94d2db0">

there used to be some issues with this but they have been resolved (see #7614)
